astropy/table/tests/test_init_table.py

```diff
@@ -519,3 +519,10 @@ def test_init_from_row_OrderedDict(table_type):
 
     with pytest.raises(ValueError):
         table_type(rows=[OrderedDict([('b', 1)]), {'a': 10, 'b': 20}])
+
+
+def test_init_from_rows_as_generator():
+    rows = [(1 + ii, 2 + ii) for ii in range(2)]
```

Comment:
I don't think this test is right: `rows` is a list here, not a generator...

Reply:
Ops! Sorry if I merged too quickly!

Reply:
@Cadair 's original use case does seemed fixed though. I'll fix the test since it is my bad to merge it too soon.

Reply:
Yes, the solution looks good, just the test is not...

Reply:
See #9316